src/transaction/transaction.spec.ts

```diff
@@ -150,6 +153,516 @@ describe('transaction', () => {
       expect(e.traces[0].message).toMatch(messageRegex)
     }
   })
+
+  describe('Cover app call inner transaction fees', async () => {
+    let appClient1: AppClient
+    let appClient2: AppClient
+    let appClient3: AppClient
+
+    beforeEach(async () => {
+      const { algorand, testAccount } = localnet.context
+      Config.configure({ populateAppCallResources: true })
+
+      const appFactory = algorand.client.getAppFactory({
+        appSpec: JSON.stringify(innerFeeContract),
+        defaultSender: testAccount,
+      })
+
+      appClient1 = (await appFactory.send.bare.create({ note: 'app1' })).appClient
+      appClient2 = (await appFactory.send.bare.create({ note: 'app2' })).appClient
+      appClient3 = (await appFactory.send.bare.create({ note: 'app3' })).appClient
+
+      appFactory.send.create
+
+      await appClient1.fundAppAccount({ amount: algo(2) })
+      await appClient2.fundAppAccount({ amount: algo(2) })
+      await appClient3.fundAppAccount({ amount: algo(2) })
+    })
+
+    test('throws when no max fee is supplied', async () => {
```

Comment:
Do we need maxFee to be required?

Reply:
Additionally, having this as a requirement means that we can use this value in the simulate transaction and we don't have to guess a value to use. If we guess too low the transaction can fail due to "fee too small", if we guess too high it can fail due to "account overspend".